src/components/expression/expression.js

```diff
@@ -172,6 +194,23 @@ genotet.ExpressionView = function(viewName, params) {
     .on('genotet.updatePanel', function() {
       this.panel.dataLoaded();
     }.bind(this));
+
+  // Set up link callbacks.
+  $(this)
+    .on('genotet.addProfile', function(event, data) {
+      var genes = /** @type {!Array<string>} */ (data);
+      genes.forEach(function(gene) {
+        var geneName = this.data.lowerGeneNames[gene];
+        var geneIndex = this.data.matrix.geneNames.indexOf(geneName);
+        var isExistent = this.data.profiles.filter(function(obj) {
+            return obj.geneName == geneName;
+          }).length != 0;
+        if (geneIndex != -1 && !isExistent) {
+          this.panel.signal('addGeneProfile', geneIndex);
```

Comment:
view does not need to send signals to loader/panel/renderer controlled by it. Call the handlers directly.